### ISSUE TITLE
Build not reproducible into a working state

### ISSUE THREAD

**Issue by stkdiretto** [IRNAS/koruza-pi#11]
I tried to build koruza-pi and ran into the following issues:

- raspbian versions from jessie onwards do not make use of the watchdog anymore but rely on systemd. The commands have changed, too, breaking the installation routine
- the python `cryptography` package requires updating `setuptools` prior to installation

I intend to rewrite the installation routine to work with newer raspbian editions, _however_ I am currently not even able to bring the koruza unit into a working state. Upon startup

- the status LED blinks
- no connection is established to the motor boards
- manually subscribing to the IPC endpoints only yields the watchdog messages, nothing more. Trying to write to the command endpoint results in an error (see below).
- manually stopping all services and setting the LED GPIOs high/low does not result in any LEDs lighting up other than the status LED

    sudo nanocat --req --connect-ipc /tmp/koruza-command.ipc --ascii --data '{"type": "command", "command": "motor_move", "next_x": 150, "next_y": 200, "next_f": 0}'
    {"message": "Motor not present.", "code": 400, "type": "cmd_error"}

**Comment by SloMusti** [comment]
The issue is likely correlated to the significant changes in may packages and this firmware is not yet maintained. We can try to provide an SD card image of the old build.

**Comment by stkdiretto** [comment]
An image would be nice, yes! The two SD cards that came with the hardware you kindly supplied were unfortunately FUBAR :(

As stated, I'd be willing to put some effort into porting the software to a current raspbian version – I just can't quite understand the internals just from following through the publicly available documentation. IMO, stopping all services and manually setting `GPIO.output(PIN_LED_LINK, GPIO.HIGH)` in Python should make the LED light up, shouldn't it? `PIN_LED_STATUS` can be toggled this way just fine, anything else just doesn't work, however.

**Comment by SloMusti** [comment]
The best effort would be to upgrade to the latest firmware of KORUZA Pro (koruza-rpi-package) and implement support for old hardware. Mainly this is configuring the right GPIOs, the only major step is sorting out the communication, or simply using the latest https://github.com/IRNAS/koruza-move-driver

**Comment by stkdiretto** [comment]
Okay, I'll give that a try. Brace for issues on that repository, though, I couldn't get it to build with raspbian stretch, as it relies on packages only available in jessie :D

**Comment by SloMusti** [comment]
Yes, we are using jessie on it. One of the best strategies which we are considering is to move the whole system into resin.io dockerized approach, which would resolve a number of issues with regards to builds and versions.

**Comment by stkdiretto** [comment]
Could you provide me with a working image so I can start with one completely working system, and try to port the other end to the rewrite?

**Comment by SloMusti** [comment]
Please find the SD card image for KORUZA V1 attached here: https://1drv.ms/u/s!AvjSCarcZuaEgVBbowK32ufX720U
Burn the image to SD card with https://etcher.io/
